src/renderer/components/settings-general-mirror.tsx

```diff
@@ -0,0 +1,114 @@
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import { AppState } from '../state';
+import {
+  Button,
+  Callout,
+  FormGroup,
+  InputGroup,
+  Menu,
+  MenuItem,
+} from '@blueprintjs/core';
+import { Popover2 } from '@blueprintjs/popover2';
+import { ELECTRON_MIRRORS } from '../mirror-constants';
+
+interface MirrorSettingsProps {
+  appState: AppState;
+}
+
+interface IMirrorSettingsState {
+  electronMirror: string;
+  electronNightlyMirror: string;
+}
+
+/**
+ * Settings electron mirror
+ *
+ * @class MirrorSettings
+ * @extends {React.Component<MirrorSettingsProps, IMirrorSettingsState>}
+ */
+@observer
+export class MirrorSettings extends React.Component<
+  MirrorSettingsProps,
+  IMirrorSettingsState
+> {
+  constructor(props: MirrorSettingsProps) {
+    super(props);
+
+    this.state = {
+      ...props.appState.electronMirrors,
+    };
+  }
+
+  private setMirror(isNightly: boolean, value: string) {
+    if (isNightly) {
+      this.setState({
+        electronNightlyMirror: value,
+      });
+      this.props.appState.electronMirrors.electronNightlyMirror = value;
+    } else {
+      this.setState({
+        electronMirror: value,
+      });
+      this.props.appState.electronMirrors.electronMirror = value;
+    }
+  }
+
+  private selectMirrorMenu(isNightly: boolean) {
+    const mirrorKey = isNightly ? 'electronNightlyMirror' : 'electronMirror';
+    return (
+      <Popover2
+        content={
+          <Menu>
+            <MenuItem
+              text="Default"
+              onClick={() => {
+                this.setMirror(isNightly, ELECTRON_MIRRORS.DEFAULT[mirrorKey]);
+              }}
+            />
+            <MenuItem
+              text="China"
+              onClick={() => {
+                this.setMirror(isNightly, ELECTRON_MIRRORS.CHINA[mirrorKey]);
+              }}
+            />
+          </Menu>
+        }
+        placement="bottom-end"
+      >
+        <Button minimal={true} rightIcon="caret-down">
+          Select {isNightly ? 'Nightly' : ''} Mirror
+        </Button>
+      </Popover2>
+    );
+  }
+
+  public render() {
+    const electronMirrorsLabel =
+      'Set the base URL of the mirror(nightly) to download from.';
+
+    return (
+      <div>
+        <h4>ELectron Mirrors</h4>
+        <Callout>
+          <FormGroup label={electronMirrorsLabel}>
+            <InputGroup
+              value={this.state.electronMirror}
+              rightElement={this.selectMirrorMenu(false)}
+              onChange={(e) => {
+                this.setMirror(false, e.currentTarget.value);
+              }}
+            />
+            <InputGroup
+              value={this.state.electronNightlyMirror}
+              rightElement={this.selectMirrorMenu(true)}
+              onChange={(e) => {
+                this.setMirror(true, e.currentTarget.value);
+              }}
+            />
```

Comment:
UX-wise I think a single selector for _both_ the nightly and regular Electron releases would be helpful. Something like this:

<img width="708" alt="image" src="https://user-images.githubusercontent.com/16010076/157985925-16e2a166-08da-48e9-b479-bccb79adbcd0.png">

We could also hide or disable the bottom two inputs unless the user chooses a "custom" mirror.

Reply:
Done. Preview Video:


https://user-images.githubusercontent.com/8198408/158007454-e01bd8db-894d-4c8b-8aa9-7277dad16b6c.mp4